libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/MapboxExtendableButtonParams.kt

```diff
@@ -0,0 +1,50 @@
+package com.mapbox.navigation.dropin
+
+import android.widget.LinearLayout
+import androidx.annotation.StyleRes
+
+/**
+ * Params used to modify button appearance.
+ *
+ * @param style style res of a button.
+ * @param layoutParams layout params of a button.
+ */
+class MapboxExtendableButtonParams(
+    @StyleRes
+    val style: Int,
+    val layoutParams: LinearLayout.LayoutParams,
+) {
+    /**
+     * Regenerate whenever a change is made
+     */
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as MapboxExtendableButtonParams
+
+        if (style != other.style) return false
+        if (layoutParams != other.layoutParams) return false
```

Comment:
Not sure if it is a problem, but `LayoutParams` classes don't implement `equals`, `hashCode` and `toString`.